### ISSUE TITLE
Cannot edit variable values in the watch window (remote 'g' packet reply is too long)

### ISSUE THREAD

**Issue by aallrd** [microsoft/vscode-cpptools#6456]
Type: Debugger

**Describe the bug**
- OS and Version: RHEL 7.6
- VS Code Version: 1.51.0 (user setup)
- C/C++ Extension Version: 1.1.0 (latest)
- Other extensions you installed (and if the issue persists after disabling them):
 - mjcrouch.perforce
 - llvm-vs-code-extensions.vscode-clangd
 - Issue persists after disabling them
- A clear and concise description of what the bug is.

I am remote debugging my binary using a target remote configuration connecting to a gdbserver.
I am trying to modify the value of a variable displayed in the debug watch window.
I double click on the value of the variable I want to modify in the watch window to enable an edit mode.
I modify the value with another one of the same type (ex, replacing 0 with 1).
I click outside of the edition placeholder or press enter to validate the value change.
It fails and the below error is displayed both in the variable value field and the debug console window:

```
<--   C (setVariable-81): {"command":"setVariable","arguments":{"variablesReference":1001,"name":"fDataDeleted","value":"1"},"type":"request","seq":81}
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (339408) <-1529-var-assign var199.fDataDeleted \"1\"\n"},"seq":9599}
1: (339408) <-1529-var-assign var199.fDataDeleted "1"
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (339410) ->1529^error,msg=\"Remote 'g' packet reply is too long (expected 560 bytes, got 816 bytes): 17995722fe7f000000000000000000000d000000000000009f985722fe7f00009f985722fe7f000017995722fe7f000030995722fe7f000000995722fe7f0000004000000000000090020000000000002a0000000000000060390eee787f0000a0a95722fe7f0000b00a00b4787f00000000000000000000200bc61400000000be90dd070000000002020000330000002b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0d90940000000000000a0d90940000000000000b0e90a407f03000000000000ffff0000787f00001e5e30ecfe7f000078ab572200000000ff0000000000ff0000000000000000ff0049444a4f42005245465f444154410045004c0050004261746368206f66206500000000000000000000000000000000000000000000000000000000000000005f5f5f0000000000436d704d61747269202020202020202020202020202020206d6d616e643e20202020202020202020000000000000000000000000000000002020202020202020202020202020203c0020202000000020202000000020200000a44e49ad48afbc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000038fafe422ee63f0000000000000000a01f0000ffffffffffffffffc0d60bd6787f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000\"\n"},"seq":9601}
1: (339410) ->1529^error,msg="Remote 'g' packet reply is too long (expected 560 bytes, got 816 bytes): 17995722fe7f000000000000000000000d000000000000009f985722fe7f00009f985722fe7f000017995722fe7f000030995722fe7f000000995722fe7f0000004000000000000090020000000000002a0000000000000060390eee787f0000a0a95722fe7f0000b00a00b4787f00000000000000000000200bc61400000000be90dd070000000002020000330000002b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0d90940000000000000a0d90940000000000000b0e90a407f03000000000000ffff0000787f00001e5e30ecfe7f000078ab572200000000ff0000000000ff0000000000000000ff0049444a4f42005245465f444154410045004c0050004261746368206f66206500000000000000000000000000000000000000000000000000000000000000005f5f5f0000000000436d704d61747269202020202020202020202020202020206d6d616e643e20202020202020202020000000000000000000000000000000002020202020202020202020202020203c0020202000000020202000000020200000a44e49ad48afbc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000038fafe422ee63f0000000000000000a01f0000ffffffffffffffffc0d60bd6787f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (339410) ->(gdb)\n"},"seq":9603}
1: (339410) ->(gdb)
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (339410) 1529: elapsed time 2\n"},"seq":9605}
1: (339410) 1529: elapsed time 2
--> R (setVariable-81): {"type":"response","request_seq":81,"success":false,"command":"setVariable","message":"Remote 'g' packet reply is too long (expected 560 bytes, got 816 bytes): 17995722fe7f000000000000000000000d000000000000009f985722fe7f00009f985722fe7f000017995722fe7f000030995722fe7f000000995722fe7f0000004000000000000090020000000000002a0000000000000060390eee787f0000a0a95722fe7f0000b00a00b4787f00000000000000000000200bc61400000000be90dd070000000002020000330000002b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0d90940000000000000a0d90940000000000000b0e90a407f03000000000000ffff0000787f00001e5e30ecfe7f000078ab572200000000ff0000000000ff0000000000000000ff0049444a4f42005245465f444154410045004c0050004261746368206f66206500000000000000000000000000000000000000000000000000000000000000005f5f5f0000000000436d704d61747269202020202020202020202020202020206d6d616e643e20202020202020202020000000000000000000000000000000002020202020202020202020202020203c0020202000000020202000000020200000a44e49ad48afbc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000038fafe422ee63f0000000000000000a01f0000ffffffffffffffffc0d60bd6787f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","body":{"error":{"id":1107,"format":"Remote 'g' packet reply is too long (expected 560 bytes, got 816 bytes): 17995722fe7f000000000000000000000d000000000000009f985722fe7f00009f985722fe7f000017995722fe7f000030995722fe7f000000995722fe7f0000004000000000000090020000000000002a0000000000000060390eee787f0000a0a95722fe7f0000b00a00b4787f00000000000000000000200bc61400000000be90dd070000000002020000330000002b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0d90940000000000000a0d90940000000000000b0e90a407f03000000000000ffff0000787f00001e5e30ecfe7f000078ab572200000000ff0000000000ff0000000000000000ff0049444a4f42005245465f444154410045004c0050004261746368206f66206500000000000000000000000000000000000000000000000000000000000000005f5f5f0000000000436d704d61747269202020202020202020202020202020206d6d616e643e20202020202020202020000000000000000000000000000000002020202020202020202020202020203c0020202000000020202000000020200000a44e49ad48afbc00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000038fafe422ee63f0000000000000000a01f0000ffffffffffffffffc0d60bd6787f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"}},"seq":9607}
```

Here is my _launch.json_:
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "debug",
            "type": "cppdbg",
            "request": "launch",
            "program": "/build/bin/prog",
            "cwd": "/src",
            "MIMode": "gdb",
            "miDebuggerPath": "/usr/bin/gdb",
            "miDebuggerServerAddress": "host:1234",
            "additionalSOLibSearchPath": "/build",
            "sourceFileMap": {
                "/SOURCE": "/src"
            },
            "setupCommands": [
                {
                    "text": "-enable-pretty-printing"
                },
                {
                    "text": "maint set worker-threads unlimited"
                },
                {
                    "text": "set sysroot /"
                }
            ],
            "logging": {
                  "trace": true,
                  "traceResponse": true,
                  "engineLogging": true
            }
        }
    ]
}
```

Do you know if setting values from the watch window possible?

**Comment by Colengms** [comment]
@WardenGnaw

**Comment by WardenGnaw** [comment]
The error: `Remote 'g' packet reply is too long` comes from GDB. 

Which version of GDB and GDB Server are you using? There seems to be reports of older gdbserver having this issue. 

Related:
https://sourceware.org/bugzilla/show_bug.cgi?id=13984

**Comment by aallrd** [comment]
I am using an internal build of the GDB 10.1 release, for both GDB and GDB server.
I read some threads relating this issue to a target architecture mismatch between the host, the target and what GDB detects but in my case everything is x86_64.
It also seems to be weirdly random (or I am missing some context), since I just tried the same setup and I was able to set the value of the variable...
Some other users in my company reported this issue to me so I am not sure what goes wrong.
